sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -41,20 +46,22 @@ public ManagedIdentityClient(ManagedIdentityClientOptions options)
 
         protected string ClientId { get; }
 
-        public virtual async ValueTask<AccessToken> AuthenticateAsync(bool async, TokenRequestContext context, CancellationToken cancellationToken)
+        public virtual async ValueTask<AccessToken> AuthenticateAsync(bool async, TokenRequestContext context,
+            CancellationToken cancellationToken)
         {
             return await _identitySource.Value.AuthenticateAsync(async, context, cancellationToken).ConfigureAwait(false);
         }
 
         private static ManagedIdentitySource SelectManagedIdentitySource(ManagedIdentityClientOptions options)
         {
-            return AppServiceV2019ManagedIdentitySource.TryCreate(options) ??
+            return
                 AppServiceV2017ManagedIdentitySource.TryCreate(options) ??
```

Comment:
I believe `AppServiceV2017ManagedIdentitySource` and `CloudShellManagedIdentitySource` should come after `AppServiceV2019ManagedIdentitySource` since App Service hosts which support `2019-08-01` will likely still set `MSI_ENDPOINT` and `MSI_SECRET` for backwards compatibility in this case we want to use the latest protocol. If we don't have a test which set's all four environment variables which app services uses `IDENTITY_ENDPOINT`, `IDENTITY_HEADER`, `MSI_ENDPOINT` and `MSI_SECRET`, and ensure we use the `AppServiceV2019ManagedIdentitySource` in this case.